wallet/src/de/schildbach/wallet/service/BlockchainStateDataProvider.kt

```
@@ -193,45 +197,47 @@ class BlockchainStateDataProvider @Inject constructor(
-                    prevBlock = blockChain.chainHead
-                }
+    override suspend fun getMasternodeAPY(): Double {
+        return withContext(Dispatchers.Default) {
+            val masternodeListManager = dashSystemService.system.masternodeListManager
+            val blockChain = dashSystemService.system.blockChain
+            if (masternodeListManager != null && blockChain != null) {
+                val mnlist = masternodeListManager.listAtChainTip
+                if (mnlist.height != 0L) {
+                    var prevBlock = try {
+                        blockChain.blockStore.get(mnlist.height.toInt() - 1)
+                    } catch (e: BlockStoreException) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)